### PR TITLE
PEO-369 Update rubocop runner

### PR DESCRIPTION
**Pull request by D3N**
[PEO-369](https://garage.atlassian.net/browse/PEO-369)

### Description

Update rubocop runner

### Before submitting the merge request make sure the following are checked

- [x] Followed the style guidelines of this project
- [x] Performed a self-review of own code
- [x] Wrote the tests that prove that fix is effective/that feature works

### CODE REVIEW

File: lib/inquisition/rubocop/rubocop_modified_runner.rb [hunk 1]
```diff
@@ -19,7 +19,7 @@ def run(paths)
       def process_file(file)
         file_started(file)
         offenses = file_offenses(file)
-        @issues << [smart_path(file), offenses] if offenses.any?
+        @issues << { smart_path(file) => offenses } if offenses.any?
```
Comment (rostikkkk2) [inline lib/inquisition/rubocop/rubocop_modified_runner.rb:20]: You can write `issues`, without @

File: lib/inquisition/rubocop/runner.rb [hunk 2]
```diff
@@ -11,21 +11,39 @@ class Runner < ::Inquisition::Runner
...
+      def choose_config
+        return USERS_CONFIG_FILE if File.exist?(USERS_CONFIG_FILE)
+
+        gem_root = File.realpath(File.join(File.dirname(__FILE__), '..', '..'))
+        File.join(gem_root, 'config', '.rubocop.yml')
+      end
+
+      def create_issue(offenses)
+        offenses.values.flatten.each do |offense|
+          @issues << Inquisition::Issue.new(level: LEVELS[offense.severity.name],
```
Comment (rostikkkk2) [inline lib/inquisition/rubocop/runner.rb]: You can write issues, without @

File: spec/inquisition/rubocop/runner_spec.rb [hunk 3]
```diff
@@ -1,3 +1,8 @@
+
+
+
+require 'pry'
```
Comment (rostikkkk2) [inline spec/inquisition/rubocop/runner_spec.rb]: Remove please

**Review by Svatok — DISMISSED** [review]

File: lib/config/.rubocop.yml [hunk 4]
```diff
@@ -1,3 +1,8 @@
+require:
```
Comment (chubchenko) [inline lib/config/.rubocop.yml]: Propose to move this file to `config/rubocop` folder and rename him to `config.yml`

File: lib/inquisition/rubocop/runner.rb [hunk 5]
```diff
@@ -11,21 +11,39 @@ class Runner < ::Inquisition::Runner
...
+      def config_store
+        config_store = ::RuboCop::ConfigStore.new
+        config_store.options_config = options[:config]
+        config_store
+      end
+
+      def choose_config
+        return USERS_CONFIG_FILE if File.exist?(USERS_CONFIG_FILE)
+
+        gem_root = File.realpath(File.join(File.dirname(__FILE__), '..', '..'))
```
Comment (chubchenko) [inline lib/inquisition/rubocop/runner.rb]: https://stackoverflow.com/questions/10132949/finding-the-gem-root

Propose to move this method to `Inquisition.root` + spec

File: lib/inquisition/rubocop/runner.rb [hunk 6]
```diff
@@ -11,21 +11,39 @@ class Runner < ::Inquisition::Runner
         fatal: :high
       }.freeze
 
-      ::RuboCop::ConfigLoader::DOTFILE = 'config/.rubocop.yml'.freeze
+      USERS_CONFIG_FILE = '.rubocop.yml'.freeze
```
Comment (chubchenko) [inline lib/inquisition/rubocop/runner.rb]: Can we use `RuboCop::ConfigLoader:: DOTFILE` instead?

File: lib/inquisition/rubocop/runner.rb [hunk 7]
```diff
@@ -11,21 +11,39 @@ class Runner < ::Inquisition::Runner
...
 
       def call
-        offenses = Inquisition::Rubocop::RuboCopModifiedRunner.new({}, ::RuboCop::ConfigStore.new).run(['.'])
+        offenses = RuboCopModifiedRunner.new(options, config_store).run(['.'])
```
Comment (chubchenko) [inline lib/inquisition/rubocop/runner.rb]: It's a little bit strange that we pass the same options to `RuboCopModifiedRunner` and `ConfigStore`

File: lib/inquisition/rubocop/runner.rb [hunk 8]
```diff
@@ -11,21 +11,39 @@ class Runner < ::Inquisition::Runner
...
-      def create_issue(offense)
-        offense_body = offense.last.first
-        @issues << Inquisition::Issue.new(severity: LEVELS[offense_body.severity.name],
-                                          path: offense.first, message: offense_body.message,
-                                          runner: self, line: offense_body.line)
+      def options
+        { config: choose_config }
+      end
+
+      def config_store
```
Comment (chubchenko) [inline lib/inquisition/rubocop/runner.rb]: Maybe would be better to create a separate class which will be responsible for the gem configuration, what do you think?

File: lib/inquisition/rubocop/runner.rb [hunk 9]
```diff
@@ -11,21 +11,39 @@ class Runner < ::Inquisition::Runner
...
+      end
+
+      def choose_config
+        return USERS_CONFIG_FILE if File.exist?(USERS_CONFIG_FILE)
+
+        gem_root = File.realpath(File.join(File.dirname(__FILE__), '..', '..'))
+        File.join(gem_root, 'config', '.rubocop.yml')
+      end
+
+      def create_issue(offenses)
```
Comment (chubchenko) [inline lib/inquisition/rubocop/runner.rb]: I think would be better to rename this method to `create_issues`, what do you think?

**Review by chubchenko — CHANGES_REQUESTED** [review]

**Pull request merged**